Gems/WhiteBox/Code/Source/SubComponentModes/EditorWhiteBoxTransformMode.cpp

```diff
@@ -6,50 +6,648 @@
+            {
+                auto vertexHandles = Api::PolygonVertexHandles(*whiteBox, polygonSelection->GetHandle());
+                DrawPoints(debugDisplay, whiteBox, worldFromLocal, viewportInfo, vertexHandles, ed_whiteBoxVertexSelection);
+                if (m_polygonIntersection.value_or(PolygonIntersection{}).GetHandle() != polygonSelection->GetHandle())
+                {
+                    DrawFace(debugDisplay, whiteBox, polygonSelection->GetHandle(), ed_whiteBoxPolygonSelection);
+                    DrawOutline(debugDisplay, whiteBox, polygonSelection->GetHandle(), ed_whiteBoxOutlineSelection);
+                }
+            }
+            else if (auto edgeSelection = AZStd::get_if<EdgeIntersection>(&m_whiteBoxSelection->m_selection))
+            {
+                auto vertexHandles = Api::EdgeVertexHandles(*whiteBox, edgeSelection->GetHandle());
+                DrawPoints(debugDisplay, whiteBox, worldFromLocal, viewportInfo, vertexHandles, ed_whiteBoxVertexSelection);
+                if (m_edgeIntersection.value_or(EdgeIntersection{}).GetHandle() != edgeSelection->GetHandle())
+                {
+                    DrawEdge(debugDisplay, whiteBox, edgeSelection->GetHandle(), ed_whiteBoxOutlineSelection);
+                }
+            }
+            else if (auto vertexSelection = AZStd::get_if<VertexIntersection>(&m_whiteBoxSelection->m_selection))
+            {
+                if (m_vertexIntersection.value_or(VertexIntersection{}).GetHandle() != vertexSelection->GetHandle())
+                {
+                    auto handles = AZStd::array<Api::VertexHandle, 1>({ vertexSelection->GetHandle() });
+                    DrawPoints(debugDisplay, whiteBox, worldFromLocal, viewportInfo, handles, ed_whiteBoxVertexSelection);
+                }
+            }
+        }
+        debugDisplay.PopMatrix();
+        debugDisplay.DepthTestOff();
     }
 
     bool TransformMode::HandleMouseInteraction(
-        [[maybe_unused]] const AzToolsFramework::ViewportInteraction::MouseInteractionEvent& mouseInteraction,
+        const AzToolsFramework::ViewportInteraction::MouseInteractionEvent& mouseInteraction,
         [[maybe_unused]] const AZ::EntityComponentIdPair& entityComponentIdPair,
-        [[maybe_unused]] const AZStd::optional<EdgeIntersection>& edgeIntersection,
-        [[maybe_unused]] const AZStd::optional<PolygonIntersection>& polygonIntersection,
-        [[maybe_unused]] const AZStd::optional<VertexIntersection>& vertexIntersection)
+        const AZStd::optional<EdgeIntersection>& edgeIntersection,
+        const AZStd::optional<PolygonIntersection>& polygonIntersection,
+        const AZStd::optional<VertexIntersection>& vertexIntersection)
     {
+        WhiteBoxMesh* whiteBox = nullptr;
+        EditorWhiteBoxComponentRequestBus::EventResult(
+            whiteBox, m_entityComponentIdPair, &EditorWhiteBoxComponentRequests::GetWhiteBoxMesh);
+
+        const bool mouseOverMainpulator = ([&]() 
```

Comment:
nit: `mouseOverMainpulator` -> `mouseOverManipulator` (also better to be explicit with captures and not use a blanked `&`) (also there are extra parens around the expression which can be removed)